src/libraries/System.Private.Xml/src/System/Xml/Core/XmlEncodedRawTextWriterAsync.cs

```diff
@@ -73,6 +73,62 @@ internal override Task WriteXmlDeclarationAsync(string xmldecl)
             return Task.CompletedTask;
         }
 
+        protected override async ValueTask DisposeAsyncCore()
+        {
+            try
+            {
+                await FlushBufferAsync().ConfigureAwait(false);
+            }
+            finally
+            {
+                // Future calls to Close or Flush shouldn't write to Stream or Writer
+                writeToNull = true;
+
+                if (stream != null)
+                {
+                    try
+                    {
+                        await stream.FlushAsync().ConfigureAwait(false);
+                    }
+                    finally
+                    {
+                        try
+                        {
+                            if (closeOutput)
+                            {
+                                await stream.DisposeAsync().ConfigureAwait(false);
+                            }
+                        }
+                        finally
+                        {
+                            stream = null;
+                        }
+                    }
+                }
+                else if (writer != null)
```

Comment:
presumably `stream != null` and `writer != null` cannot occur at the same time?

Reply:
yes there would be underlying stream or writer to write, but not both. Checking them both for null here is just to make sure they are not already disposed